main.tf

```diff
@@ -0,0 +1,73 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "4.58.0"
```

Comment:
Version locking is a good and healthy practice.
Why? :)

Reply:
If we don't use version locking, it's very likely that not the same version will be fetched every time we run the template. A new version might come out that is not yet compatible with other providers we use in our templates, which may cause bugs. By specifying which version we will be using, we ensure the code template runs correctly each time, and prevent future bugs.